### ISSUE TITLE
Update progress?

### ISSUE THREAD

**Issue by FrozenAssassine** [FrozenAssassine/EasePass#179]
What is the progress in case of the next update?

From my side, most of the stuff is fixed!
Do you have other stuff we should fix before release, if not we should do some testing and finish the update?
 
@finn-freitag @Flamifly

**Comment by Flamifly** [comment]
From my side the stuff is fixed also.
The only thing I will do is checking if everything is translated etc.

**Comment by FrozenAssassine** [comment]
Awesome, I just have to fix the import and export dialogs, but then I am finished too.

**Comment by Flamifly** [comment]
Is there a target date?

**Comment by FrozenAssassine** [comment]
> Is there a target date?

Not really, when everything is done, we all do some testing and when we fixed all those bugs. I write the changelog, maybe do some more screenshots for the store page and then I create the package and publish. This process takes as long as needed...

**Comment by Flamifly** [comment]
Did I broke the Chart or was that one of you?
![image](https://github.com/user-attachments/assets/18896c8d-cb1d-438e-8bec-38c23c42a176)

The Password keeps being leaked

**Comment by FrozenAssassine** [comment]
That was me 😂

**Comment by Flamifly** [comment]
Phew
If it would have been me I wouldn't know I messed that up😂

**Comment by finn-freitag** [comment]
@FrozenAssassine  is known for changing other people's code and breaking it. 😉

**Comment by FrozenAssassine** [comment]
@finn-freitag Because I have to fix your "interesting" implemented code

**Comment by Flamifly** [comment]
Never change a running System my man

**Comment by Flamifly** [comment]
And if you fix the Password chart thing could you also fix the Translation? :smile: 
![image](https://github.com/user-attachments/assets/4f5840e5-26c6-44d8-b37d-d448f3d8dd0f)

**Comment by FrozenAssassine** [comment]
Got everything fixed.

**Comment by FrozenAssassine** [comment]
I think there is not much left until the update is done. 
I would be happy about a short info from you @finn-freitag and @Flamifly, how it looks and if you think the same as I do.  Also if there are some things that still need to be done, tell me, so we can finish it.

**Comment by Flamifly** [comment]
There is one thing at the Moment about the Password Generation.
Is there a reason why we check if a Password has Punctuations?
Don't get me wrong I understand that you want to check if a Password includes special chars.

But for example if a Password includes <>%€@ etc. it wont be true because it is not a Punctuation
Should that be changed?

**Comment by FrozenAssassine** [comment]
Oh sorry, you already asked that somewhere else. I totally forgot to answer, my bad😢 
But you are right, this is not correct and should be fixed. Good you noticed 🥇

**Comment by Flamifly** [comment]
I can fix that

**Comment by Flamifly** [comment]
Should I change the Translation also?

**Comment by FrozenAssassine** [comment]
It's probably the best so yes

**Comment by Flamifly** [comment]
The Sorting for the Password strength does not work properly
So I think the Passwords strength should be equal to the green chart entries (If I have all entries green the first Item should have the same amount and the next one the same or less (desc)
In my case I have some with 7 green entries and 8 entries before and after

**Comment by Flamifly** [comment]
If I Export a Database with a different Password does it ask me to enter the Password for the new Database or the current?

**Comment by FrozenAssassine** [comment]
> The Sorting for the Password strength does not work properly So I think the Passwords strength should be equal to the green chart entries (If I have all entries green the first Item should have the same amount and the next one the same or less (desc) In my case I have some with 7 green entries and 8 entries before and after
> 
> (Hope that makes sense what I say)

![image](https://github.com/user-attachments/assets/f8b5c25f-7400-4b4b-a275-848e447f5978)
I found the problem... A little copy-paste error. I take that on me😂. I will fix and commit.

**Comment by FrozenAssassine** [comment]
> If I Export a Database with a different Password does it ask me to enter the Password for the new Database or the current?

Surely for the new database. The current database is already loaded so no need to enter the password again. You just have to enter any password, which will be used to encrypt the new database.

**Comment by Flamifly** [comment]
Also if you click on the Eye (show Password) it will show always the Password even if you switch to another one is that on purpose?

**Comment by FrozenAssassine** [comment]
I don't really know, it is not really a bug, but also not some kind of feature. 
Finn and I already thought about this, but I think it is better to always hide the password, when changing the page. Probably saver.

**Comment by Flamifly** [comment]
If I am in the DatabaseManagement and click on another Database enter password and click ok
It will load the Database without letting me know does that need to change?

If I import a Database it won't load the imported Database automaticly

**Comment by FrozenAssassine** [comment]
The first thing you mentioned is not correct currently. The database should only be selected, when the user presses the button to do that. I will have a look at it too.
 
 When importing, you enter the password to be able to import it, after that the database should be "unlocked" and the user has access to the buttons on the right.

**Comment by Flamifly** [comment]
> The first thing you mentioned is not correct currently. The database should only be selected, when the user presses the button to do that. I will have a look at it too.
> 
> When importing, you enter the password to be able to import it, after that the database should be "unlocked" and the user has access to the buttons on the right.

Yes I would expect that if we want to keep the Situation 1. that it would be the same for the second.

**Comment by FrozenAssassine** [comment]
Can you have a look at #186

**Comment by Flamifly** [comment]
yea wait

**Comment by Flamifly** [comment]
Seems fine to me and works as expected 👍🏻

**Comment by Flamifly** [comment]
I do not have anything else that has to be done now
I tested I think everything

**Comment by FrozenAssassine** [comment]
Awesome, very good job. I will also take another look tomorrow for some final tests. Also I will test stuff like coming from old version and checking whether the transition works fine etc.

**Comment by Flamifly** [comment]
@FrozenAssassine I saw your changes on the CopyTextBox and noticed that there is a
Startswith("http") why do we check if it startswith http and not https?

Also while I was texting my Plugin I created Tons of Passwords and at my Website Textbox a http:// was added even if I had https:// already set. Was the Problem on my side or not?

**Comment by FrozenAssassine** [comment]
I did not write this code, I just changed it. We need @finn-freitag here. But we need a better way for that. You have an idea?

**Comment by Flamifly** [comment]
For what exactly?
For the default adding http://?

**Comment by FrozenAssassine** [comment]
Exactly that

**Comment by Flamifly** [comment]
I would check if it startseith http:// or if it starts with https:// and add by default https://

**Comment by Flamifly** [comment]
> Also while I was texting my Plugin I created Tons of Passwords and at my Website Textbox a http:// was added even if I had https:// already set. Was the Problem on my side or not?

This error was my bad because I added a " " to my samples as first char

**Comment by finn-freitag** [comment]
> @FrozenAssassine I saw your changes on the CopyTextBox and noticed that there is a Startswith("http") why do we check if it startswith http and not https?
> 
> Also while I was texting my Plugin I created Tons of Passwords and at my Website Textbox a http:// was added even if I had https:// already set. Was the Problem on my side or not?

I have made it so that http:// is always appended, because if the website is actually used with https://, normal browsers automatically redirect the http:// to https://. Conversely, this is not always the case.

**Comment by Flamifly** [comment]
> I have made it so that http:// is always appended, because if the website is actually used with https://, normal browsers automatically redirect the http:// to https://. Conversely, this is not always the case.

Thanks for the explentation
I would Change the Logic a bit
To check if it starts with http:// or https:// to be 100% sure that the url does not be like http://httpmyweb.com and the User does not add the http://. This probably will like never Happen just to be really sure

**Comment by Flamifly** [comment]
Should this ChangeLog be deleted?
https://github.com/FrozenAssassine/EasePass/blob/master/changelog.md

**Comment by FrozenAssassine** [comment]
> Should this ChangeLog be deleted? https://github.com/FrozenAssassine/EasePass/blob/master/changelog.md

Lol this thing still exists😂.
I will remove this

**Comment by Flamifly** [comment]
Seems like it should not😂

**Comment by FrozenAssassine** [comment]
I guess it's finally done so should I start building the app from the current state of source code or are there any things that need to change beforehand?
@finn-freitag @Flamifly?

**Comment by Flamifly** [comment]
I don't think so

**Comment by FrozenAssassine** [comment]
![image](https://github.com/user-attachments/assets/96800a41-0406-4ae4-b353-491a7747d2f5)

Well well well, I think we did it. Big thanks to you @Flamifly for your help so far. If you don't mind we can add on Discord for simpler communication of future projects idk? If you want you can email me. Email is on my Github main page

**Comment by Flamifly** [comment]
No need to thank it's nice to help🙂
I emailed you for my DC
